Makefile

```diff
@@ -44,37 +103,104 @@ ci: check-tidy test e2e-test
 
 .PHONY: start
 start:
-	go run ./cmd/server/main.go
+	$(COMPILER_FLAGS) go run ./cmd/main.go
 
 .PHONY: start-local
 start-local:
 	rm -rf db/
 	rm -rf metrics/data/
-	go run cmd/main.go run \
-		--flow-network-id=flow-emulator \
-		--coinbase=FACF71692421039876a5BB4F10EF7A439D8ef61E \
-		--coa-address=f8d6e0586b0a20c7 \
-		--coa-key=2619878f0e2ff438d17835c2a4561cb87b4d24d72d12ec34569acd0dd4af7c21 \
-		--wallet-api-key=2619878f0e2ff438d17835c2a4561cb87b4d24d72d12ec34569acd0dd4af7c21 \
-		--coa-resource-create=true \
-		--gas-price=0 \
-		--log-writer=console \
-		--profiler-enabled=true \
-		--profiler-port=6060
+	$(COMPILER_FLAGS) go run cmd/main.go run $(EMULATOR_ARGS)
 
 # Use this after running `make build`, to test out the binary
 .PHONY: start-local-bin
 start-local-bin:
 	rm -rf db/
 	rm -rf metrics/data/
-	./flow-evm-gateway run \
-		--flow-network-id=flow-emulator \
-		--coinbase=FACF71692421039876a5BB4F10EF7A439D8ef61E \
-		--coa-address=f8d6e0586b0a20c7 \
-		--coa-key=2619878f0e2ff438d17835c2a4561cb87b4d24d72d12ec34569acd0dd4af7c21 \
-		--wallet-api-key=2619878f0e2ff438d17835c2a4561cb87b4d24d72d12ec34569acd0dd4af7c21 \
-		--coa-resource-create=true \
-		--gas-price=0 \
-		--log-writer=console \
-		--profiler-enabled=true \
-		--profiler-port=6060
+	$(COMPILER_FLAGS) go run cmd/main.go run $(EMULATOR_ARGS)
+
+# Build docker image from local sources
+.PHONY: docker-build-local
+docker-build-local:
+ifdef GOARCH
+	$(eval ARCH=$(GOARCH))
+endif
+	docker build --build-arg ARCH=$(ARCH) --no-cache -f dev/Dockerfile -t "$(CONTAINER_REGISTRY)/evm-gateway:$(COMMIT)" .
+
+# Docker run for local development
+.PHONY: docker-run-local
+docker-run-local:
+	flow emulator &
+	sleep 3
+
+	$(call check_and_append,coinbase,EMULATOR_COINBASE)
+	$(call check_and_append,coa-address,EMULATOR_COA_ADDRESS)
+	$(call check_and_append,coa-key,EMULATOR_COA_KEY)
+
+	$(eval CMD_ARGS += --flow-network-id=flow-emulator --log-level=debug --gas-price=0 --log-writer=console --profiler-enabled=true)
+
+	docker run -p 8545:8545 --add-host=host.docker.internal:host-gateway "$(CONTAINER_REGISTRY)/evm-gateway:$(COMMIT)" $(CMD_ARGS)
+
+
+# Build docker image for release
+.PHONY: docker-build
+docker-build:
+ifdef GOARCH
+	$(eval ARCH=$(GOARCH))
+endif
+	docker build --build-arg VERSION="$(VERSION)" --build-arg ARCH=$(ARCH)  -f Dockerfile -t "$(CONTAINER_REGISTRY)/evm-gateway:$(IMAGE_TAG)" \
```

Comment:
@vishalchangrani - let me know what should be done for tags - i copied what we do for other Flow nodes, but not sure that is what you need